ansibullbot/triagers/plugins/docs_info.py

```diff
@@ -0,0 +1,306 @@
+                        if ds_span is not None:
+                            func_map.ds_line_start = len(
+                                content[:ds_span.start()].splitlines()
+                            )
+                            func_map.ds_line_end = (
+                                func_map.ds_line_start + len(func_doc.splitlines())
+                        )
+                    class_map.funcs.append(func_map)
+
+            mod_map.classes.append(class_map)
+
+    return mod_map
+
+def _check_py_changes(file_content, diff):
+    """ Check a python file's changes to see if they're only docstring
+        changes.
+    """
+
+    diff = _get_diff_info(diff)
+    source = None
+
+    if file_content is not None:
+        source = _get_ast_info(file_content)
+
+    if source is not None:
+        for line in diff:
+            # Check if change applies to module DOCUMENTATION (if exists)
+            if source.doc_string:
+                if source.ds_line_start <= line["lineno"] <= source.ds_line_end:
+                    continue
+
+            # Check if change applies to module EXAMPLES (if exists)
+            if source.example_string:
+                if source.ex_line_start <= line["lineno"] <= source.ex_line_end:
+                    continue
+
+            # Find appropriate class if it exists
+            source_class = source.find_class(line["lineno"])
+            if source_class is not None:
+                if source_class.doc_string:
+                    if source_class.ds_line_start <= line["lineno"] <= source_class.ds_line_end:
+                        continue
+
+                source_func = source_class.find_function(line["lineno"])
+                if source_func is not None and source_func.doc_string:
+                    if source_func.ds_line_start <= line["lineno"] <= source_func.ds_line_end:
+                        continue
+
+            # If we made it this far, this change is outside docs/examples
+            return False
+        else:
+            return True
+
+def get_docs_facts(iw):
+    """ Cycle through the files and triage if changes only
+        affect docs-related files/content.
+    """
+    dfacts = {
+        "is_docs_only": False
+    }
+
+    if not iw.is_pullrequest():
+        return dfacts
+
+    docs_only = True
+
+    for commit in iw.commits:
+        commit_files = iw.get_commit_files(commit)
+        if commit_files is None:
+            # "Sorry, this diff is temporarily unavailable due to heavy server load."
+            # Preserve docsite label to prevent potential waffling
+            dfacts["is_docs_only"] = "docsite" in iw.labels
+            return dfacts
+
+        for changed_file in commit_files:
+            if isinstance(changed_file, dict):
+                changed_file = CommitFile(changed_file)
+
+            if _is_docs_path(changed_file.filename):
+                continue
+            else:
```

Comment:
All these `else` are not really needed since there are `continue` or `break` used.